### ISSUE TITLE
Feature request: get reasons behind predictions made by Decision Trees

### ISSUE THREAD

**Issue by WladdGorshenin** [dotnet/machinelearning#913]
My team had long conversation with ML.NET's team (based on issue #599) and we came up with this feature request as the result.

For a model based on decision trees we'd like to have public API method similar to Scikit Learn's "decision_path" which does the following: for each input sample in outputs route traveled from root to leaf in each tree (i.e. list of nodes and edges).

We need that to understand reasons behind each prediction our model made.

**Comment by lefig** [comment]
Absolutely. My team is looking into feature analysis upfront by way of selection as part of analysis. To discount model fields. Doing something like...

for train, test in ss:
        # obtain the index of each feature on the training set
        idx,_,_ = CIFE.cife(X[train], y[train], n_selected_features=num_fea)

        # obtain the dataset on the selected features
        features = X[:, idx[0:num_fea]]

        # train a classification model with the selected features on the training dataset
        clf.fit(features[train], y[train])

        # predict the class labels of test data
        y_predict = clf.predict(features[test])

        # obtain the classification accuracy on the test data
        acc = accuracy_score(y[test], y_predict)
        correct = correct + acc

But it would be great to incorporate this process within the pipeline.

**Comment by Zruty0** [comment]
We have a component called 'what-the-feature' in our internal codebase. It appears that this will solve the oirginal problem.

**Comment by GalOshri** [comment]
Hi @WladdGorshenin, from a few other discussions it sounded like the "what-the-feature"/"per-example feature gains" will solve the original problem. Can we close this issue and track moving these components to ML.NET in #599?
